File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include<iostream>
#include<vector>
using namespace std;
int check[201];
void dfs(vector<vector<int>> computers, int n, int visited){
    check[visited] = 1;
  	for(int j=0; j<n; j++) { 
    	if(check[j] == 0 && computers[visited][j] == 1) {   
      		dfs(computers,n,j);
        }
    } 
}

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    for(int i =0; i < n; i++){
        if(check[i] == 0){
         	dfs(computers, n, i);
            answer++;
        }
    }
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visited = new boolean[n];
        for(int i = 0 ; i < n; i++){
            if(visited[i] == false){
                dfs(computers, i, visited);
                answer ++;
            }    
        }
        return answer;
    }
    
    static void dfs(int[][] computers, int node , boolean[] visited){
        visited[node] = true;
        for(int i = 0 ; i < computers.length; i++){
            if(computers[node][i] == 1 && visited[i] == false){
                dfs(computers, i, visited);
            }
        }
    }
}